Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -141,6 +142,40 @@ inline void ElementRuleCollector::addElementStyleProperties(const StylePropertie
     addMatchedProperties(WTFMove(matchedProperty), DeclarationOrigin::Author);
 }
 
+void ElementRuleCollector::collectMatchingRules(CascadeLevel level)
+{
+    std::optional<MatchRequest> matchRequest;
+    switch (level) {
+    case CascadeLevel::Author:
+        matchRequest.emplace(m_authorStyle);
+        break;
+
+    case CascadeLevel::User:
+        if (m_userStyle)
+            matchRequest.emplace(*m_userStyle);
+        break;
+
+    case CascadeLevel::UserAgent:
+        ASSERT_NOT_REACHED();
+        return;
+    }
+
+    if (matchRequest)
+        collectMatchingRules(*matchRequest);
```

Comment:
you could just call collectMatchingRules from the switch without emplace tricks.

Reply:
LOL good point not sure why i kept this so complicated 🤣